### DISCUSSION TITLE
Notifcation Controller: Support Microsoft Power Automate Workflow as a Provider

### DISCUSSION THREAD

**Discussion by CelPyn** [fluxcd/flux2#4880] [Proposals]
Microsoft is [deprecating support of Office 365 connectors](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/).

> We will gradually roll out this change in waves:
> - Wave 1 – effective August 15th, 2024: All new Connector creation will be blocked within all clouds
> - Wave 2 – effective October 1st, 2024: All connectors within all clouds will stop working

It's hard to find documentation on this, but it seems like the [Connector Card](https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-reference#connector-card-for-microsoft-365-groups) API contract, (which the current `msteams` provider implementation uses), is not supported by the available "incoming webhook" action in Power Automate. Instead, it requires the [Adaptive Card](https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-reference#adaptive-card) API contract, which in turn is not supported by the Connector webhook.

With this discussion, I propose the creation of a new `Provider` Kind which supports the Adaptive Card.

**Comment by zeeZ** [comment]
Big point from the comments (according to the counter there are 125, but the other pages don't load) that a lot of people might be running into, before it gets missed:

**Sending a message in private channels isn't supported.**

**Reply by CelPyn** [reply]
Hi, I tested this out in my org's Teams tenant and managed to get this working.

It's a bit of a dirty workaround, but if you set the `Post As` value to `User` it will post a message under your name, which uses your permissions and will allow the Workflow to send messages to private channels that you're part of on your behalf.

I'm assuming that you can also use `Flow Bot` once you add the Workflows app to your Teams channel, but I'm currently not able to get that to work. The app gets installed twice and the Workflows configured via this way are not showing up. Running the workflow with this value results in a "Can't install the Power Automate app automatically" error.

This is probably out of scope for the implementation of this hypothetical Provider, but I do agree that this would need to be clearly documented, at least until Microsoft fixes things on their end.

**Reply by jornh** [reply]
Related issue: https://github.com/fluxcd/notification-controller/issues/878
